src/main/java/com/avispl/symphony/dal/communicator/lumen/vc/tr60a/LumenVCTR60ACommunicator.java

```diff
@@ -0,0 +1,1985 @@
+	 * <li>WB Mode</li>
+	 * <li>RGain</li>
+	 * <li>BGain</li>
+	 * <li>WB OnePushTrigger</li>
+	 *
+	 * @param value is the value of controllable property
+	 * @param commandField is the command get from controllable property name
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 */
+	private void whiteBalanceControl(String value, Command commandField, Map<String, String>stats, List<AdvancedControllableProperty>advancedControllableProperties) {
+		try {
+			switch (commandField){
+				case WB_MODE:
+					performControl(PayloadCategory.CAMERA, commandField, WBMode.getByName(value).getCode());
+					break;
+				case RGAIN:
+					float levelRed = Float.parseFloat(value);
+					byte[] red = convertToTwoBytePQ(levelRed, 127);
+					performControl(PayloadCategory.CAMERA, commandField, (byte) 0x00, (byte) 0x00, red[0], red[1]);
+					handlePictureSlider(commandField, Command.RGAIN_CURRENT, value, LumenVCTR60AConstants.ZERO, "127", LumenVCTR60AConstants.RANGE_START_HUE_LEVEL, 127F, stats, advancedControllableProperties);
+					break;
+				case BGAIN:
+					float levelBlue = Float.parseFloat(value);
+					byte[] blue = convertToTwoBytePQ(levelBlue, 127);
+					performControl(PayloadCategory.CAMERA, commandField, (byte) 0x00, (byte) 0x00, blue[0], blue[1]);
+					handlePictureSlider(commandField, Command.BGAIN_CURRENT, value, LumenVCTR60AConstants.ZERO, "127", LumenVCTR60AConstants.RANGE_START_HUE_LEVEL, 127F, stats, advancedControllableProperties);
+					break;
+				case WB_ONE_PUSH_TRIGGER: {
+					performControl(PayloadCategory.CAMERA, Command.WB_ONE_PUSH_TRIGGER);
+					break;
+				}
+				default: {
+					throw new IllegalStateException("Unexpected value: " + commandField);
+				}
+			}
+		} catch (Exception e){
+			logger.error("error during command " + commandField.getName() + " send", e);
+			throw new IllegalStateException("Error while sending command " + commandField.getName());
+		}
+	}
+
+	/**
+	 * This method used to perform control of all properties by send, receive command from device
+	 *
+	 * @param payloadCategory is the category of payload of the command to be sent
+	 * @param command is the command to be sent
+	 * @param param is the param of command to be sent
+	 */
+	public void performControl(PayloadCategory payloadCategory, Command command, byte... param) {
+		byte[] request;
+		byte[] response;
+
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			request = buildSendPacket(cameraIDInt, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), payloadCategory.getCode(),
+					command.getCode(), param);
+			response = send(request);
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			this.logger.error("error during command " + command.getName() + " send", e);
+			if(!Command.PRESET.getName().equals(command.getName())){
```

Comment:
it's fine to have it like this in here though, since IllegalStateException is conditional. Original exception reference should be passed though